### DISCUSSION TITLE
Using the tilde symbol "~" as a remapping

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#152] [Ideas]
@andreivladbrg what do you think about remapping the `src` folder with the tilde symbol `~`?

```text
# remappings.txt
~/=src/
```

We are currently using `@sablier/v2-core` but I think that it can be a bit confusing to import the source contracts using the full package name from within the package itself.

**Comment by andreivladbrg** [comment]
I agree, it can be confusing.

Did you know that we can import  in the `test` folder any file from the `src` folder  like this :
```
import { SablierV2Linear } from "src/SablierV2Linear.sol";
```
Without adding anything in the remappings file?

Personally I would prefer to use the default foundry settings.

**Reply by PaulRBerg** [reply]
No, I didn't know that you can use `src` directly, but I think this is also confusing. And if Foundry ever decides to change the default, we (or a future colleague of ours) will have to investigate what happened - and it might not be super obvious that the problem is Foundry.

For the sake of explicitness, and also succinctness, I suggest we use the tilde symbol `~`. This is also what we're using in the front-end repo, and so we would have "parity" in import styles between our repositories.

**Comment by PaulRBerg** [comment]
@andreivladbrg any new thoughts on this?

I would really like to do this.

**Reply by andreivladbrg** [reply]
IMO using "~" is confusing and not using the default Foundry configurations, because this symbol is used for the root.

If you really like this, go ahead, I will adapt to your preferences.

**Reply by PaulRBerg** [reply]
What do you mean by "this symbol is used for the root"?

Tagging @razgraf to confirm that we are currently using `~` in `v2-interfaces` to refer to the `src` directory of every app.

**Reply by razgraf** [reply]
In our v2-interfaces the `~` symbol is used as an imaginary root of the package tree. It's not mapping 1:1 to any package, it's just simplifying path resolution.

Examples:
```json
     "~/client/*": ["./apps/client/src/*"],
     "~/components/*": ["./packages/components/src/*"]
```

**Comment by PaulRBerg** [comment]
I [tweeted](https://twitter.com/PaulRBerg/status/1593694139211579392) about this and the responses I got have made change my mind and go with @andreivladbrg's suggested approach of using `src` rather than `~`.

Locking.
